src/tests.rs

```diff
@@ -48,3 +49,11 @@ fn vec_exposes_init_err() {
     #[allow(unused_imports)]
     use crate::vec::InitError;
 }
+
+#[test]
+fn should_be_able_to_recover_memory_from_memory_manager() {
```

Comment:
Can we also add a test where there's an underlying reference and `into_memory` returns `None`?

Reply:
Certainly.  Done.